src/main.c

```diff
@@ -247,10 +344,85 @@ void WriteInterfaces()
     free(InterfaceArray);
 }
 
+void CreateVmSwitchPacketFragment(PEVENT_RECORD ev)
+{
+    wchar_t Buffer[8192];
+    PROPERTY_DATA_DESCRIPTOR Desc;
+    int Err;
+
+    // The SourcePortId will be used to create interface
+    Desc.PropertyName = (unsigned long long)L"SourcePortId";
+    Desc.ArrayIndex = ULONG_MAX;
+    Err = TdhGetProperty(ev, 0, NULL, 1, &Desc, sizeof(VMSwitchPacketFragment.VmSwitchSourcePortId), (PBYTE)&VMSwitchPacketFragment.VmSwitchSourcePortId);
+    if (Err != NO_ERROR) {
+        printf("TdhGetProperty VmSwitchSourcePortId failed with %u\n", Err);
+        return;
+    }
+
+    // SourceNicName
+    char SourceNicName[1024];
+    Desc.PropertyName = (unsigned long long)(L"SourceNicName");
+    Desc.ArrayIndex = ULONG_MAX;
+    ULONG ParamNameSize = 0;
+    (void)TdhGetPropertySize(ev, 0, NULL, 1, &Desc, &ParamNameSize);
+    Err = TdhGetProperty(ev, 0, NULL, 1, &Desc, sizeof(Buffer), (PBYTE)Buffer);
+    if (Err != NO_ERROR) {
+        Buffer[0] = L'\0';
+    }
+    Buffer[ParamNameSize / sizeof(wchar_t) + 1] = L'\0';
+    WideCharToMultiByte(CP_ACP, 0, Buffer, -1, SourceNicName, ARRAYSIZE(SourceNicName), NULL, NULL);
+    SourceNicName[wcslen(Buffer)] = '\0';
+    memset(Buffer, 0, sizeof(Buffer));
+    VMSwitchPacketFragment.VmSwitchSourceNicName = malloc(sizeof(char) * ParamNameSize + 1);
```

Comment:
More malloc calls in the hot path

Reply:
I know sorry but I don't have really another option here, as the char* length cannot be known in advance. 
The property's size has to retrieve first... Eventually, I can remove one malloc() call as SourceNicName is not used.... 
I can allocate char arrays big enough in VMSWITCH_PACKET_FRAGMENT struct, but this is waste for memory...

Reply:
There's a pattern you can use here and for the other "not sure how much memory is needed" situations I've commented on: maintain a global dynamically-allocated buffer, and each time you need memory, if the buffer is large enough then use it; otherwise free it and allocate a new buffer that's large enough. This way, unless [very unlikely] there is a slightly larger required buffer size for handling each event, there will be very few total malloc calls.

Reply:
Please check the last commit, I have drastically reduced the number of malloc() calls by only allocating memory during pass1, and for non-existing interfaces. Basically, only needed memory is allocated. 
This is now O(3n) when n is the number of interfaces. Before it was more O(2m) where m is the number of events...

For example, with one Hyper-V traces / 700MB :
- malloc calls numbers were before 1715643 (2m), and now is 798(3n)

Reply:
You mentioned in the commit description that the runtime went from 75sec to 55sec on a 700MB trace with this optimization, which is good validation.

What is the runtime on the same system and file without this PR?

Reply:
Resolved offline- the perf numbers sound good to me after these optimizations.